cmd/deviceservice/update/update.go

```diff
@@ -0,0 +1,149 @@
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/metadata"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/urlclient/local"
+	"github.com/edgexfoundry/go-mod-core-contracts/models"
+
+	"github.com/spf13/cobra"
+)
+
+const DeviceSrviceTempl = `{
+   "Id": "{{.Id}}",
+   "Name" : "{{.Name}}",
+   "Description" : "{{.Description}}",
+   "AdminState" : "{{.AdminState}}",
+   "OperatingState" : "{{.OperatingState}}",
+   "Labels" : [
+    {{- $labelsLenght := len .Labels}}
+    {{- range $idx, $l := .Labels}}
+      "{{$l}}" {{if not (lastElem $idx $labelsLenght)}},{{end}} 
+    {{- end}}
+    ],
+   "Addressable" :
+   {
+     "Name" : "{{.Addressable.Name}}"
+   }
+ }`
+
+var name string
+var file string
+
+// NewCommand returns the update device service command
+func NewCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "update",
+		Short: "Update device service",
+		Long:  `Update device service(s) described in the given JSON file or use the interactive mode with additional flags.`,
+		RunE:  deviceServiceHandler,
+	}
+	cmd.Flags().StringVarP(&name, "name", "n", "", "Name")
+	cmd.Flags().StringVarP(&file, "file", "f", "", "Json file containing device service configuration to update")
+	return cmd
+}
+
+func deviceServiceHandler(cmd *cobra.Command, args []string) error {
+	if name != "" && file != "" {
+		return errors.New("DeviceService could be updated by providing a file, or by specifying device service name to be updated and using interactive mode. ")
+	}
+
+	if name == "" && file == "" {
+		return errors.New("Please, provide file or device service name ")
+	}
+
+	if file != "" {
+		return updateDeviceServiceFromFile()
+	}
+
+	updatedDeviceService, err := parseDeviceService(name)
+	if err != nil {
+		return err
+	}
+
+	client := local.New(config.Conf.Clients["Metadata"].Url() + clients.ApiDeviceServiceRoute)
+	err = metadata.NewDeviceServiceClient(client).Update(context.Background(), updatedDeviceService)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+//parseDeviceService loads a device service to be updated and open a default editor for customization
+func parseDeviceService(name string) (models.DeviceService, error) {
+	var err error
+	client := local.New(config.Conf.Clients["Metadata"].Url() + clients.ApiDeviceServiceRoute)
+	ds, err := metadata.NewDeviceServiceClient(client).DeviceServiceForName(context.Background(), name)
+	if err != nil {
+		return models.DeviceService{}, err
+	}
+
+	updatedDeviceServiceBytes, err := editor.OpenInteractiveEditor(ds, DeviceSrviceTempl, template.FuncMap{
+		"lastElem": editor.IsLastElementOfSlice,
+	})
+
+	if err != nil {
+		return models.DeviceService{}, err
+	}
+	var updatedDeviceService models.DeviceService
+	err = json.Unmarshal(updatedDeviceServiceBytes, &updatedDeviceService)
+	if err != nil {
+		return models.DeviceService{}, errors.New("Unable to execute the command. The provided information is not valid: " + err.Error())
```

Comment:
not important, but instead of "not valid" can use "invalid"

Reply:
done